### PR TITLE
Add compileTimeOnly annotation in 2.10

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: core/src/main/scala_2.10/macrocompat/annotations.scala [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package scala.annotation
+
+import scala.annotation.meta._
+
+@getter @setter @beanGetter @beanSetter @companionClass @companionMethod
```
Comment (milessabin) [inline core/src/main/scala_2.10/macrocompat/annotations.scala:21]: Huh?

Comment (ghost) [inline core/src/main/scala_2.10/macrocompat/annotations.scala:21]: see 
https://github.com/scala/scala/blob/27da46343cd545534819300235bc64ab74958c92/src/library/scala/annotation/compileTimeOnly.scala#L21

Comment (milessabin) [inline core/src/main/scala_2.10/macrocompat/annotations.scala:21]: OK, ignore that :-)

**Comment by milessabin** [comment]
LGTM ... many thanks, merging :-)

**Pull request merged**